File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_backend/entity/inventory/SeedBatchEntity.java
package com.example.fieldsync_inventory_backend.entity.inventory;

import jakarta.persistence.*;
import lombok.*;
import com.example.fieldsync_inventory_backend.entity.Auditable;

import java.time.Instant;

@Entity
@Table(name = "seed_batch", uniqueConstraints = @UniqueConstraint(columnNames = {
        "variety_id", "year", "season_id", "grading", "generation_id", "germination"
}))
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SeedBatchEntity extends Auditable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "variety_id")
    private RiceVarietyEntity variety;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "season_id")
    private SeasonEntity season;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "generation_id")
    private RiceGenerationEntity generation;

    @Column(nullable = false)
    private Integer year;

    @Column(nullable = false)
    private Boolean grading;

    @Column(nullable = false)
    private Boolean germination;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/local/entity/SeedBatchEntity.kt
package com.example.fieldsync_inventory_app.data.local.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "seed_batch")
data class SeedBatchEntity(
    @PrimaryKey val id: Long, // Use the ID from the backend as the primary key
    @ColumnInfo(name = "variety_id")
    val varietyId: Int,
    val varietyName: String,
    @ColumnInfo(name = "generation_id")
    val generationId: Int,
    val generationName: String,
    @ColumnInfo(name = "season_id")
    val seasonId: Int,
    val seasonName: String,
    val year: Int,
    val grading: Boolean,
    val germination: Boolean,
    val description: String?,
    @ColumnInfo(name = "created_at")
    val createdAt: Long,
    @ColumnInfo(name = "updated_at")
    val updatedAt: Long,
    @ColumnInfo(name = "deleted_at")
    val deletedAt: Long?
)